===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(field_cb:str)) at 33 (line 268, column 7),
        Insert: if (true) free(*(field_cb:str)) at 44 (line 249, column 15) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/recutils-1.8/utils/csv2rec.c
+++ b/table1/benchmarks/recutils-1.8/utils/csv2rec.c
@@ -248,6 +248,7 @@ field_cb (void *s, size_t len, void *data)
         {
           if (ctx->num_fields > ctx->num_field_names)
             {
+              free(str);
               char *source = csv2rec_csv_file;
 
               if (!source)
@@ -266,6 +267,7 @@ field_cb (void *s, size_t len, void *data)
         }
 
       ctx->num_fields++;
+      free(str);
     }
 }
 
